synapse/handlers/message.py

```diff
@@ -776,8 +800,16 @@ async def create_and_send_nonmember_event(
                     assert event.internal_metadata.stream_ordering
                     return event, event.internal_metadata.stream_ordering
 
+            prev_events = event_dict.get("prev_events")
```

Comment:
Can we pass these in instead? I'm wary of accidentally letting clients set it.

Reply:
Good idea :+1: